### ISSUE TITLE
production crash on some devices   java.lang.NullPointerException

### ISSUE THREAD

**Issue by levepic** [gbumps/react-native-screenguard#29]
Still receiving 

` Fatal Exception: java.lang.NullPointerException
java.util.Objects.requireNonNull (Objects.java:203)
com.screenguard.ScreenGuardModule.lambda$deactivateShield$2$com-screenguard-ScreenGuardModule (ScreenGuardModule.java:101) `

using 0.3.0-stable

Usage:

`
function initScreenGuard() {
    try {
        ScreenGuardModule.register(
            //insert any hex color you want here, default black if null or empty
            '#FF0000',
            () => {
            alert("Screenshots are not allowed!");
        });
    }
    catch (e) {
        console.warn("ScreenGuardModule.register", e);
    }
}

useEffect(() => {
        initScreenGuard();
        console.log("initScreenGuard()");
        return () => {
            ScreenGuardModule.unregister();
            console.log("ScreenGuardModule.unregister()");
        };        
    }, []);
`

**Comment by gbumps** [comment]
Thanks for reporting. I understand now. This happened when app jump from background to foreground => the context is null and has not been reinitialized yet, and when deactivating, this must be done on ui thread.

At instance, you can do a temporary fix on native side, while I revise my code again:

```
@ReactMethod
    public void deactivateShield() {
        try {
            if (mHandlerBlockScreenShot != null &&
                    getReactApplicationContext().getCurrentActivity() != null
            ) {
                mHandlerBlockScreenShot.post(() -> Objects.requireNonNull(
                        getReactApplicationContext().getCurrentActivity()
                ).getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE));
                mHandlerBlockScreenShot = null;
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```

**Comment by levepic** [comment]
With the provided modifications the following error appeared:

`Fatal Exception: java.lang.NullPointerException:
       at java.util.Objects.requireNonNull(Objects.java:220)
       at com.screenguard.ScreenGuardModule.lambda$activateShield$1$com-screenguard-ScreenGuardModule(ScreenGuardModule.java:73)
       at com.screenguard.ScreenGuardModule$$ExternalSyntheticLambda1.run(:2)
       at android.os.Handler.handleCallback(Handler.java:942)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8757)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)`

Line 73 is:
mHandlerBlockScreenShot.post(() -> Objects.requireNonNull(

of activateShield method

Im not sure if I could reproduce the error on emulator, it havent happened there.

**Comment by gbumps** [comment]
yes, please do the same with the `deactivateShield` as well and recheck carefully before applying this fix for next fix

```
@ReactMethod
    public void activateShield() {
        try {
            if (mHandlerBlockScreenShot == null) {
                mHandlerBlockScreenShot = new Handler(Looper.getMainLooper());
            }
            if (getReactApplicationContext().getCurrentActivity() != null) {
                 mHandlerBlockScreenShot.post(() -> Objects.requireNonNull(
                    getReactApplicationContext().getCurrentActivity()
            ).getWindow().setFlags(
                    WindowManager.LayoutParams.FLAG_SECURE,
                    WindowManager.LayoutParams.FLAG_SECURE
            ));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```

**Comment by levepic** [comment]
The error is now
`Fatal Exception: com.facebook.react.bridge.NativeArgumentsParseException: ScreenGuard.activateShield got 1 arguments, expected 0`
although I inserted String hexColor in public void activateShield(String hexColor)
Without hexcolor the error is a build error bc of the missing param

**Comment by gbumps** [comment]
Mark as done, closing. Feel free to reopen again if problem occurs. A fix for this issue will be delivered shortly 😉.

**Comment by levepic** [comment]
Thank you! Seems OK in the fresh realase! Waiting for more data from firebase to be sure. I'll push it to another app with larger user base if it looks stable for a good while. That will probably reveal any hidden bugs if there is any, hopefully none:)

**Comment by levepic** [comment]
No crashes! The fix definitely works!

**Comment by gbumps** [comment]
glad to hear that. If you have any questions,  don't hesitate to ask or create PR for fixing and improvement, i'm very appreciated!.

**Comment by gbumps** [comment]
This issue has been resolved at version 0.3.2, please install and rebuild. Thanks!

**Comment by gbumps** [comment]
> Having some crashes with 0.3.2-stable, a few percent of users, usually at logout, probably when leaving the app and unregister happens:
> 
> 100% Android 11 Device states 100% background
> 
> ` Fatal Exception: java.lang.NullPointerException: at java.util.Objects.requireNonNull(Objects.java:220) at com.screenguard.ScreenGuardModule.lambda$deactivateShield$2$com-screenguard-ScreenGuardModule(ScreenGuardModule.java:107) at com.screenguard.ScreenGuardModule$$ExternalSyntheticLambda2.run(:2) at android.os.Handler.handleCallback(Handler.java:938) at android.os.Handler.dispatchMessage(Handler.java:99) at android.os.Looper.loop(Looper.java:236) at android.app.ActivityThread.main(ActivityThread.java:8057) at java.lang.reflect.Method.invoke(Method.java) at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
> 
> ` App.js:
> 
> function initScreenGuard() { try { ScreenGuardModule.register( //insert any hex color you want here, default black if null or empty '#000000', () => { alert("Screenshots are not allowed!"); }); } catch (e) { console.warn("ScreenGuardModule.register", e); } }
> 
> const App = () => {
> 
> useEffect(() => { initScreenGuard(); console.log("initScreenGuard()"); return () => { ScreenGuardModule.unregister(); console.log("ScreenGuardModule.unregister()"); }; }, []);`

**Comment by gbumps** [comment]
It could be somehow when calling `deactivateShield` at native, at the 1st time call `getReactApplicationContext().getCurrentActivity()`, this is still not null yet. Unluckily, at 2nd call after checking, it became null -> crash.

Relax, the problem is only a few percent and the gap between those call `getReactApplicationContext().getCurrentActivity()`  too little, but I still provide a quick fix for you if you want to:

```
@ReactMethod
    public void activateShield(String hexColor) {
        try {
            if (mHandlerBlockScreenShot == null) {
                mHandlerBlockScreenShot = new Handler(Looper.getMainLooper());
            }
            if (currentReactContext == null) {
                currentReactContext = getReactApplicationContext();
            }
            Activity currentActivity = currentReactContext.getCurrentActivity();
            if (currentActivity != null) {
                mHandlerBlockScreenShot.post(() ->
                                currentActivity.getWindow().setFlags(
                        WindowManager.LayoutParams.FLAG_SECURE, 
                        WindowManager.LayoutParams.FLAG_SECURE
                ));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

@ReactMethod
    public void deactivateShield() {
        try {
            if (mHandlerBlockScreenShot == null) {
                mHandlerBlockScreenShot = new Handler(Looper.getMainLooper());
            }
            if (currentReactContext == null) {
                currentReactContext = getReactApplicationContext();
            }
            Activity currentActivity = currentReactContext.getCurrentActivity();
                if (currentActivity != null) {
                   mHandlerBlockScreenShot.post(() ->
                           currentActivity
                     .getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE));
                    mHandlerBlockScreenShot = null;
                } else {
                    Log.w("ACTIVITY_SCREENSHOT", "handler is null");
                }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```

**Comment by levepic** [comment]
The fix is working, no crashes so far! Also being tested on ios in live app.

**Comment by gbumps** [comment]
Thanks. As this problem is too small to make an update instantly after `v0.3.2`, I'll mark this as an improvement and update alongside with new feature.

**Comment by levepic** [comment]
Just to note: no crashes so far either on ios/android with the fixed version, using it on larger active userbase!

**Comment by gbumps** [comment]
hi @levepic, this issue has been completely resolved at version `v0.3.5`, please reinstall and rebuild. 

Thanks!
